File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.py
N, M = map(int,input().split())
arr = [list(input()) for _ in range(N)]
ans = []
for r in range(N-7):
    for c in range(M-7):
        w_cnt = 0
        b_cnt = 0
        for i in range(r,r+8):
            for j in range(c,c+8):
                if (i+j)%2 == 0:
                    if arr[i][j] != 'W':
                        w_cnt += 1
                    else:
                        b_cnt += 1
                else:
                    if arr[i][j] != 'B':
                        w_cnt += 1
                    else:
                        b_cnt += 1
        ans.append(w_cnt)
        ans.append(b_cnt)

print(min(ans))

File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.js
let input = require("fs").readFileSync("dev/stdin").toString().trim().split("\n");

let [N, M] = input[0].split(" ").map(Number);

let board = input.slice(1, N + 1).map((row) => row.trim().split(""));
let answer = [];

for (r = 0; r < N - 7; r++) {
  for (c = 0; c < M - 7; c++) {
    let w_cnt = 0;
    let b_cnt = 0;
    for (i = r; i < r + 8; i++) {
      for (j = c; j < c + 8; j++) {
        if ((i + j) % 2 === 0) {
          if (board[i][j] !== "W") {
            w_cnt += 1;
          } else {
            b_cnt += 1;
          }
        } else {
          if (board[i][j] !== "B") {
            w_cnt += 1;
          } else {
            b_cnt += 1;
          }
        }
      }
    }
    answer.push(w_cnt);
    answer.push(b_cnt);
  }
}

console.log(Math.min(...answer));
